### DISCUSSION TITLE
Train Gaussian Mixture model

### DISCUSSION THREAD

**Discussion by ljpgood** [zach401/acnportal#104] [Q&A]
Hi，@sunash，I now run your 1.2Comparing Infrastructure Designs using ACN-Sim experiment ，but the error is founded in  “ gmm = pickle.load(open(r'C:\Users\ljp\Desktop\data\jpl_weekday_40.pkl', 'rb'))‘’,    like this :gmm = pickle.load(open(r'C:\Users\ljp\Desktop\data\jpl_weekday_40.pkl', 'rb'))
_pickle.UnpicklingError: invalid load key, '\xef'.  Could you tell me how to solve it?
Also I want to know how trans a another Gaussian Mixture Model ，could you tell me some details? 
It would be best if you could give me the code.Thank you very much.

**Comment by sunash** [comment]
Hi @ljpgood, while we work on this, please try running this notebook in google colab. Go to https://colab.research.google.com/ go to the Github option and type in the URL of the notebook you want to run. Let me know if you run into any issues.
 
@zach401 can give you info on training new GMMs.

**Reply by ljpgood** [reply]
hi,@sunash,thank you  for your reply.while I run your 1.2Comparing Infrastructure Designs using ACN-Sim experiment in google colab.
![1](https://user-images.githubusercontent.com/104896016/170040361-73ed852f-0074-43b4-82a2-36f63fbf3cf9.PNG)
.This  error  is founded in my computer,too.I suspect the model jpl_weekday_40.pkl is damaged.Could you help me solve it?

**Reply by zach401** [reply]
This problem you encountered is due to a version issue. The pickled model was using sklearn 0.20.2 while Colab now uses version 1+. 

You can downgrade the sklearn version by running. 

```
import sklearn 

if sklearn.__version__ != "0.20.2":
    subprocess.check_call([sys.executable, "-m", "pip", "uninstall", "scikit-learn", "-y"])
    subprocess.check_call([sys.executable, "-m", "pip", "install", "scikit-learn==0.20.2"])
```

You will need to then restart the runtime, but click `Runtime > Restart Runtime` on the top toolbar.

**Reply by zach401** [reply]
For training GMMs, I've written a short Colab which includes code snippets for using `sklearn` directly or using the `GuassianMixtureEvents` utility in `acnportal`. The later is helpful if you want to create events for `acnsim` simulations. 

https://colab.research.google.com/drive/17mgJCuipKUEWg7O1kv9Is-qYm405mz4_?usp=sharing
